### PR TITLE
feat(rpc) Implement WalletSignMessage

### LINKED ISSUE #4510: WalletSignMessage

### PR DESCRIPTION
**Pull request by lemmih**
## Summary of changes

Changes introduced in this pull request:

- Implement `WalletSignMessage` - this signs the CID of a message and returns a `SignedMessage`.

## Reference issue to close (if applicable)

Closes #4510

## Change checklist

- [x] I have performed a self-review of my own code,
- [x] I have made corresponding changes to the documentation,
- [x] I have added tests that prove my fix is effective or that my feature works (if possible),
- [x] I have made sure the [CHANGELOG][1] is up-to-date. All user-facing changes should be reflected in this document.

[1]: https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md

### CODE REVIEW

File: src/rpc/methods/wallet.rs [hunk 1]
```diff
@@ -223,6 +225,50 @@ impl RpcMethod<2> for WalletSign {
...
+            Err(_) => {
+                let key_info = crate::key_management::try_find(&key_addr, keystore)?;
+                Key::try_from(key_info)?
+            }
+        };
+
+        let sig = crate::key_management::sign(
+            *key.key_info.key_type(),
+            key.key_info.private_key(),
+            message.cid().unwrap().to_bytes().as_slice(),
```
Comment (aatifsyed) [inline src/rpc/methods/wallet.rs:260]: I'm surprised that you're ok with panicking in this handler

File: src/rpc/methods/wallet.rs [hunk 2]
```diff
@@ -223,6 +225,50 @@ impl RpcMethod<2> for WalletSign {
...
+        )?;
+
+        // Could use `SignedMessage::new_unchecked` here but let's make sure
+        // we're actually signing the message as expected.
+        let smsg = SignedMessage::new_from_parts(message, sig)
+            .expect("Internal error: Generated incorrect message signature");
```
Comment (aatifsyed) [inline src/rpc/methods/wallet.rs:266]: should bail instead

**Review by aatifsyed — APPROVED** [review]

File: src/rpc/methods/wallet.rs (hunk 1 above)
Comment (lemmih) [reply src/rpc/methods/wallet.rs:260]: I copy-pasted the signing code, and I didn't have the mental fortitude to fix the root cause. Hashing a message should be infallible but untangling that spaghetti will be rough.

File: src/rpc/methods/wallet.rs (hunk 2 above)
Comment (lemmih) [reply src/rpc/methods/wallet.rs:266]: I don't think so. Mind explaining why?

File: src/rpc/methods/wallet.rs (hunk 1 above)
Comment (aatifsyed) [reply src/rpc/methods/wallet.rs:260]: I think I'd prefer an `expect("<here's why this is impossible>")` or a ticket link instead :)

File: src/rpc/methods/wallet.rs (hunk 2 above)
Comment (aatifsyed) [reply src/rpc/methods/wallet.rs:266]: Same as above :)

File: src/rpc/methods/wallet.rs (hunk 1 above)
Comment (lemmih) [reply src/rpc/methods/wallet.rs]: Done.

**Review by hanabi1224 — APPROVED** [review]

**Review by aatifsyed — APPROVED** [review]

**Review by sudo-shashank — APPROVED** [review]

**Review by LesnyRumcajs — COMMENTED** [review]
Do we lack any infrastructure code to add identity tests? I mean, we already have tests for signing blocks, so everything should be in place, no?

**Review by LesnyRumcajs — APPROVED** [review]

**Comment by lemmih** [comment]
> Do we lack any infrastructure code to add identity tests? I mean, we already have tests for signing blocks, so everything should be in place, no?

Nice! Our testing framework is even better than I thought. :)

**Pull request merged**